### ISSUE TITLE
No method for pm_next!

### ISSUE THREAD

**Issue by itsdfish** [TuringLang/AdvancedHMC.jl#163]
Hi-

I recently encountered an error after upgrading to v0.2.0. Here is a MWE: 

```
using Turing, Random, Distributions

@model model(N, k) = begin
    θ ~ Beta(1, 1)
    k ~ Binomial(N, θ)
end

Random.seed!(10125)
N = 10
k = rand(Binomial(N, .5))
chain = sample(model(N, k), NUTS(1000,.8), 2000)
```

Error:

```
julia> chain = sample(model(N, k), NUTS(1000,.8), 2000)
┌ Info: Found initial step size
└   ϵ = 3.2
ERROR: MethodError: no method matching pm_next!(::ProgressMeter.Progress, ::NamedTuple{(:n_steps, :is_accept, :acceptance_rate, :log_density, :hamiltonian_energy, :hamiltonian_energy_error, :max_hamiltonian_energy_error, :tree_depth, :numerical_error, :step_size, :nom_step_size),Tuple{Int64,Bool,Float64,Float64,Float64,Float64,Float64,Int64,Bool,Float64,Float64}}, ::Int64, ::AdvancedHMC.Adaptation.DiagEuclideanMetric{Float64,Array{Float64,1}})
Closest candidates are:
  pm_next!(::Any, ::NamedTuple) at /home/dfish/.julia/packages/AdvancedHMC/Sn9Ek/src/sampler.jl:64
Stacktrace:
 [1] #callback#38(::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(AbstractMCMC.callback), ::Random._GLOBAL_RNG, ::DynamicPPL.Model{var"##inner_function#584#21",NamedTuple{(:N, :k),Tuple{Int64,Int64}},DynamicPPL.ModelGen{(:N, :k),var"###model#598",NamedTuple{(),Tuple{}}},Val{()}}, ::DynamicPPL.Sampler{NUTS{Turing.Core.ForwardDiffAD{40},(),AdvancedHMC.Adaptation.DiagEuclideanMetric},Turing.Inference.HMCState{DynamicPPL.VarInfo{NamedTuple{(:θ,),Tuple{DynamicPPL.Metadata{Dict{DynamicPPL.VarName{:θ},Int64},Array{Beta{Float64},1},Array{DynamicPPL.VarName{:θ},1},Array{Float64,1},Array{Set{DynamicPPL.Selector},1}}}},Float64},AdvancedHMC.NUTS{AdvancedHMC.MultinomialTS,AdvancedHMC.GeneralisedNoUTurn,AdvancedHMC.Leapfrog{Float64},Float64},AdvancedHMC.Adaptation.StanHMCAdaptor{AdvancedHMC.Adaptation.DiagPreconditioner{Array{Float64,1},AdvancedHMC.Adaptation.WelfordVar{Array{Float64,1}}},AdvancedHMC.Adaptation.NesterovDualAveraging{Float64}},AdvancedHMC.PhasePoint{Array{Float64,1},AdvancedHMC.DualValue{Float64,Array{Float64,1}}}}}, ::Int64, ::Int64, ::Turing.Inference.HamiltonianTransition{NamedTuple{(:θ,),Tuple{Tuple{Array{Float64,1},Array{String,1}}}},NamedTuple{(:n_steps, :is_accept, :acceptance_rate, :log_density, :hamiltonian_energy, :hamiltonian_energy_error, :max_hamiltonian_energy_error, :tree_depth, :numerical_error, :step_size, :nom_step_size),Tuple{Int64,Bool,Float64,Float64,Float64,Float64,Float64,Int64,Bool,Float64,Float64}},Float64}, ::Turing.Inference.HMCCallback{ProgressMeter.Progress}) at /home/dfish/.julia/packages/Turing/a0nSz/src/inference/hmc.jl:588
 [2] callback(::Random._GLOBAL_RNG, ::DynamicPPL.Model{var"##inner_function#584#21",NamedTuple{(:N, :k),Tuple{Int64,Int64}},DynamicPPL.ModelGen{(:N, :k),var"###model#598",NamedTuple{(),Tuple{}}},Val{()}}, ::DynamicPPL.Sampler{NUTS{Turing.Core.ForwardDiffAD{40},(),AdvancedHMC.Adaptation.DiagEuclideanMetric},Turing.Inference.HMCState{DynamicPPL.VarInfo{NamedTuple{(:θ,),Tuple{DynamicPPL.Metadata{Dict{DynamicPPL.VarName{:θ},Int64},Array{Beta{Float64},1},Array{DynamicPPL.VarName{:θ},1},Array{Float64,1},Array{Set{DynamicPPL.Selector},1}}}},Float64},AdvancedHMC.NUTS{AdvancedHMC.MultinomialTS,AdvancedHMC.GeneralisedNoUTurn,AdvancedHMC.Leapfrog{Float64},Float64},AdvancedHMC.Adaptation.StanHMCAdaptor{AdvancedHMC.Adaptation.DiagPreconditioner{Array{Float64,1},AdvancedHMC.Adaptation.WelfordVar{Array{Float64,1}}},AdvancedHMC.Adaptation.NesterovDualAveraging{Float64}},AdvancedHMC.PhasePoint{Array{Float64,1},AdvancedHMC.DualValue{Float64,Array{Float64,1}}}}}, ::Int64, ::Int64, ::Turing.Inference.HamiltonianTransition{NamedTuple{(:θ,),Tuple{Tuple{Array{Float64,1},Array{String,1}}}},NamedTuple{(:n_steps, :is_accept, :acceptance_rate, :log_density, :hamiltonian_energy, :hamiltonian_energy_error, :max_hamiltonian_energy_error, :tree_depth, :numerical_error, :step_size, :nom_step_size),Tuple{Int64,Bool,Float64,Float64,Float64,Float64,Float64,Int64,Bool,Float64,Float64}},Float64}, ::Turing.Inference.HMCCallback{ProgressMeter.Progress}) at /home/dfish/.julia/packages/Turing/a0nSz/src/inference/hmc.jl:588
 [3] #sample#6(::Bool, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(sample), ::Random._GLOBAL_RNG, ::DynamicPPL.Model{var"##inner_function#584#21",NamedTuple{(:N, :k),Tuple{Int64,Int64}},DynamicPPL.ModelGen{(:N, :k),var"###model#598",NamedTuple{(),Tuple{}}},Val{()}}, ::DynamicPPL.Sampler{NUTS{Turing.Core.ForwardDiffAD{40},(),AdvancedHMC.Adaptation.DiagEuclideanMetric},Turing.Inference.HMCState{DynamicPPL.VarInfo{NamedTuple{(:θ,),Tuple{DynamicPPL.Metadata{Dict{DynamicPPL.VarName{:θ},Int64},Array{Beta{Float64},1},Array{DynamicPPL.VarName{:θ},1},Array{Float64,1},Array{Set{DynamicPPL.Selector},1}}}},Float64},AdvancedHMC.NUTS{AdvancedHMC.MultinomialTS,AdvancedHMC.GeneralisedNoUTurn,AdvancedHMC.Leapfrog{Float64},Float64},AdvancedHMC.Adaptation.StanHMCAdaptor{AdvancedHMC.Adaptation.DiagPreconditioner{Array{Float64,1},AdvancedHMC.Adaptation.WelfordVar{Array{Float64,1}}},AdvancedHMC.Adaptation.NesterovDualAveraging{Float64}},AdvancedHMC.PhasePoint{Array{Float64,1},AdvancedHMC.DualValue{Float64,Array{Float64,1}}}}}, ::Int64) at /home/dfish/.julia/packages/AbstractMCMC/pi3j6/src/AbstractMCMC.jl:223
 [4] #sample at ./none:0 [inlined]
 [5] #sample#5 at /home/dfish/.julia/packages/AbstractMCMC/pi3j6/src/AbstractMCMC.jl:194 [inlined]
 [6] (::StatsBase.var"#kw##sample")(::NamedTuple{(:progress,),Tuple{Bool}}, ::typeof(sample), ::DynamicPPL.Model{var"##inner_function#584#21",NamedTuple{(:N, :k),Tuple{Int64,Int64}},DynamicPPL.ModelGen{(:N, :k),var"###model#598",NamedTuple{(),Tuple{}}},Val{()}}, ::DynamicPPL.Sampler{NUTS{Turing.Core.ForwardDiffAD{40},(),AdvancedHMC.Adaptation.DiagEuclideanMetric},Turing.Inference.HMCState{DynamicPPL.VarInfo{NamedTuple{(:θ,),Tuple{DynamicPPL.Metadata{Dict{DynamicPPL.VarName{:θ},Int64},Array{Beta{Float64},1},Array{DynamicPPL.VarName{:θ},1},Array{Float64,1},Array{Set{DynamicPPL.Selector},1}}}},Float64},AdvancedHMC.NUTS{AdvancedHMC.MultinomialTS,AdvancedHMC.GeneralisedNoUTurn,AdvancedHMC.Leapfrog{Float64},Float64},AdvancedHMC.Adaptation.StanHMCAdaptor{AdvancedHMC.Adaptation.DiagPreconditioner{Array{Float64,1},AdvancedHMC.Adaptation.WelfordVar{Array{Float64,1}}},AdvancedHMC.Adaptation.NesterovDualAveraging{Float64}},AdvancedHMC.PhasePoint{Array{Float64,1},AdvancedHMC.DualValue{Float64,Array{Float64,1}}}}}, ::Int64) at ./none:0
 [7] #sample#2(::Nothing, ::Base.Iterators.Pairs{Union{},Union{},Tuple{},NamedTuple{(),Tuple{}}}, ::typeof(sample), ::DynamicPPL.Model{var"##inner_function#584#21",NamedTuple{(:N, :k),Tuple{Int64,Int64}},DynamicPPL.ModelGen{(:N, :k),var"###model#598",NamedTuple{(),Tuple{}}},Val{()}}, ::NUTS{Turing.Core.ForwardDiffAD{40},(),AdvancedHMC.Adaptation.DiagEuclideanMetric}, ::Int64) at /home/dfish/.julia/packages/Turing/a0nSz/src/inference/Inference.jl:159
 [8] sample(::DynamicPPL.Model{var"##inner_function#584#21",NamedTuple{(:N, :k),Tuple{Int64,Int64}},DynamicPPL.ModelGen{(:N, :k),var"###model#598",NamedTuple{(),Tuple{}}},Val{()}}, ::NUTS{Turing.Core.ForwardDiffAD{40},(),AdvancedHMC.Adaptation.DiagEuclideanMetric}, ::Int64) at /home/dfish/.julia/packages/Turing/a0nSz/src/inference/Inference.jl:158
 [9] top-level scope at none:0

```
 System info:

Julia 1.3.1

```
(v1.3) pkg> st Turing
    Status `~/.julia/environments/v1.3/Project.toml`
    [80f14c24] AbstractMCMC v0.1.0
    [0bf59076] AdvancedHMC v0.2.20
    [31c24e10] Distributions v0.21.12
    [c7f686f2] MCMCChains v1.0.0
    [4c63d2b9] StatsFuns v0.9.3
    [fce5fe82] Turing v0.8.1

(v1.3) pkg> st AdvancedHMC
    Status `~/.julia/environments/v1.3/Project.toml`
    [0bf59076] AdvancedHMC v0.2.20
    [d96e819e] Parameters v0.12.0
    [2913bbd2] StatsBase v0.32.0
    [4c63d2b9] StatsFuns v0.9.3
```

**Comment by xukai92** [comment]
Sorry I forget to make a new release for Turing. Fixing it now.

**Comment by xukai92** [comment]
This should be fixed now with Turing v0.8.1.
